specification/specs/common.ts

```diff
@@ -13,27 +13,31 @@ type TimeSpan = string
 interface SourceDocument {}
 @class_serializer("ErrorCauseFormatter")
 class ErrorCause {
-  additional_properties: Dictionary<string, any>;
+  shard: Union<integer, string>;
+  stack_trace: string;
+  type: string;
+
   bytes_limit: long;
   bytes_wanted: long;
   caused_by: ErrorCause;
   column: integer;
+  col: integer;
   failed_shards: ShardFailure[];
   grouped: boolean;
   index: string;
   index_uuid: string;
   language: string;
   licensed_expired_feature: string;
   line: integer;
+  max_buckets: integer;
   phase: string;
   reason: string;
   resource_id: string[];
+  'resource.id': string;
```

Comment:
There is more then one road to ~~rome~~ exceptions see:

https://github.com/elastic/elasticsearch/issues/27672


We need to annotate these two as exposed under the same property but for the type valiation work they have to exist separately.

Ultimately the refactor here is that additional properties have to go on a dictionary on error e.g `error.properties['resource.id']` instead which is what the linked issue also suggests.

Reply:
Ah ok, I found [where it's output](https://github.com/elastic/elasticsearch/blob/cb99487eb0a60f2c6c4a2b87b97cb3e453ced2b6/server/src/main/java/org/elasticsearch/ElasticsearchException.java#L335-L337). The keys in the metadata map are `es.something` and the `es.` prefix is stripped out on serialization.

> We need to annotate these two as exposed under the same property

Do you mean it's either `resource_id` or `resource.id` (I couldn't find `resource_id` in 8.x)? We can represent this with an annotation, e.g.:

```
@renamed("8.0.0", "resource.id")
resource_id: string[];
'resource.id': string
```

But that's out of scope for this PR

Reply:
Was thinking 
```
/**
* description
* @aliases resource.id
*/
resource_id: string[];
```
 
In general the model compiler is moving away from annotations in favor for jsdocs. Jsdocs can appear everywhere, annotations are limited where they can be placed.

Reply:
This makes me sad that TS doesn't support annotations everywhere. Having this captured as strongly typed first-class language features would be awesome. But since this is not available, indeed it's better to have only one way to capture annotations.

We have however to include annotation checks in the spec parser to capture typos in annotation names and validate their values. I opened https://github.com/elastic/elastic-client-generator/issues/45 for this (plus any other validation we may want to add)